### PR TITLE
Update the installer!

### PR DESCRIPTION
**Pull request by pschmitt**
- Update default install dir to XDG_DATA_HOME/zinit/zinit.git (https://github.com/zdharma-continuum/zinit/issues/40)
- Shellchecked the install script
- Updated references to zdharma, the wiki etc.
- Bonus: One can now set the branch to use for installs via `ZINIT_BRANCH` (defaults to master) This might be useful for development, CI and when (if!) we decide to update change the main branch of zinit to "main".

### CODE REVIEW

File: doc/install.sh [hunk 1]
```diff
@@ -64,19 +64,19 @@ fi
...
-if egrep '(zinit|zplugin)\.zsh' "$THE_ZDOTDIR/.zshrc" >/dev/null 2>&1; then
+if grep -E '(zinit|zplugin)\.zsh' "$THE_ZDOTDIR/.zshrc" >/dev/null 2>&1; then
     echo "[34m▓▒░[0m .zshrc already contains \`zinit …' commands – not making changes."
     RCUPDATE=0
 fi
 
 if [ $RCUPDATE -eq 1 ]; then
     echo "[34m▓▒░[0m Updating $THE_ZDOTDIR/.zshrc (10 lines of code, at the bottom)"
-    ZINIT_HOME="$(echo $ZINIT_HOME | sed "s|$HOME|\$HOME|")"
+    ZINIT_HOME="$(echo "$ZINIT_HOME" | sed "s|$HOME|\$HOME|")"
```
Comment (docwhat) [inline doc/install.sh:74]: This can be done entirely in Zsh

File: doc/install.sh [hunk 2]
```diff
@@ -6,47 +6,47 @@
...
 
 ZINIT_BIN_DIR_NAME="${ZINIT_BIN_DIR_NAME:-$ZPLG_BIN_DIR_NAME}"
 if [ -z "$ZINIT_BIN_DIR_NAME" ]; then
-    ZINIT_BIN_DIR_NAME="bin"
+    ZINIT_BIN_DIR_NAME="zinit.git"
 fi
 
 if ! test -d "$ZINIT_HOME"; then
-    mkdir "$ZINIT_HOME"
+    mkdir -p "$ZINIT_HOME"
```
[the remark this answers is not in the document]
Comment (pschmitt) [reply doc/install.sh:18]: It's actually needed here.

$HOME/.local/share may not exist.

File: doc/install.sh [hunk 3]
```diff
@@ -6,47 +6,47 @@
...
+    echo "[1;31m▓▒░[0m Something went wrong: [1;32mgit[0m is not available, cannot proceed."
     exit 1
 fi
 
+ZINIT_BRANCH="${ZINIT_BRANCH:-master}"
+
 # Get the download-progress bar tool
+GIT_PROCESS_SCRIPT_URL="https://raw.githubusercontent.com/zdharma-continuum/zinit/${ZINIT_BRANCH}/git-process-output.zsh"
+mkdir -p /tmp/zinit
+cd /tmp/zinit || { echo "Failed to cd to /tmp/zinit" >&2; exit 1; }
```
[the remark this answers is not in the document]
Comment (pschmitt) [reply doc/install.sh:33]: Makes sense to do so but a bit out of scope for this MR.

File: doc/install.sh (hunk 1 above)
Comment (pschmitt) [reply doc/install.sh:74]: Yeah the entire script could in zsh. What's your point?

File: doc/install.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (pschmitt) [reply doc/install.sh]: Done: f4e1468f33f90786a5f973bca4ea10d05fb79b3e

**Review by alichtman — APPROVED** [review]
Lgtm

**Comment by pschmitt** [comment]
> Lgtm. Would love to see bash coloring functions instead of inlined escape characters, but it's just a wishlist item.

That actually bothered me as well. Didn't want to create a "I REWROTE THE ENTIRE INSTALLER PLZ REVIEW" MR though :D

I'll create an issue to refactor and over-engineer the install script.

File: doc/install.sh (hunk 1 above)
Comment (docwhat) [reply doc/install.sh:83]: Ah, I missed that it was in `/bin/sh`.  My bad.

**Pull request merged**